core/src/mender-client.c

```diff
@@ -926,21 +930,21 @@ mender_client_update_work_function(void) {
  *
  * mender_update_module is guaranteed be not NULL since the first
  * successful transition (from the DOWNLOAD state). */
-#define NEXT_STATE                                                                               \
-    if (MENDER_OK == ret) {                                                                      \
-        update_state = update_state_transitions[update_state].success;                           \
-        assert(NULL != mender_update_module);                                                    \
-        mender_log_debug("Entering state %s", update_state_str[update_state]);                   \
-        mender_storage_save_update_state(update_state, mender_update_module->artifact_type);     \
-        ret = MENDER_OK;                                                                         \
-    } else {                                                                                     \
-        update_state = update_state_transitions[update_state].failure;                           \
-        mender_log_debug("Entering state %s", update_state_str[update_state]);                   \
-        if (NULL != mender_update_module) {                                                      \
-            mender_storage_save_update_state(update_state, mender_update_module->artifact_type); \
-        }                                                                                        \
-        ret = MENDER_OK;                                                                         \
-        continue;                                                                                \
+#define NEXT_STATE                                                                         \
+    if (MENDER_OK == ret) {                                                                \
+        update_state = update_state_transitions[update_state].success;                     \
+        assert(NULL != mender_update_module);                                              \
+        mender_log_debug("Entering state %s", update_state_str[update_state]);             \
+        mender_deployment_data_set_state(mender_client_deployment_data, update_state);     \
```

Comment:
The previous one used to write to non-volatile storage. This one only writes to volatile memory. Is this taken care of?

Reply:
I believe so, it's stored in `mender_client_deployment_data` which is written to the store on `MENDER_UPDATE_STATE_REBOOT`

Reply:
Right, so should we just write the deployment data each time we save the state? Right now the deployment data is only written on `MENDER_UPDATE_STATE_REBOOT`, but should we write it each time the state is set?